### PR TITLE
 Challenge Jest Hands-on

### PR DESCRIPTION
**Pull request by nyattsu72**
## Outline
Jest assignment of study session held at MorikenJuku.
[honds-on/jest](https://github.com/kenmori/handsonFrontend/tree/master/jest)

## Task
- [x] Task1
   Add script ("test": "jest") `to package.json and describe so that the test runs with" npm test ".
  Written 3 test codes
- [x] Task2
  Check if it is a numerical value.
  Check if the numbers in the string have been converted to numbers.
- [x] Task3
  Check if there is orange text in the array.

### CODE REVIEW

File: index.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { exportAllDeclaration } from '@babel/types';
+import { TestScheduler } from 'jest';
```
Comment (kenmori) [inline index.test.js]: remove

File: index.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { exportAllDeclaration } from '@babel/types';
```
Comment (kenmori) [inline index.test.js]: remove

File: index.test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	//Check if it text
+	test('Check if the string has been converted to an integer (decimal)', () => {
+		const string = '123456789';
+		expect(toNumber(string)).toBe(123456789);
+	});
+});
+
+describe('filterOrange', () => {
+	test('Check if there is orange text in the array.', () => {
```
Comment (kenmori) [inline index.test.js]:
```suggestion
	test('Check if there is "orange" element  in the array.', () => {
```

File: index.test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		const arr = ['orange', 'apple', 'peach'];
+		const argument = 'orange';
+		expect(new Set(arr)).toContain('orange');
```
Comment (kenmori) [inline index.test.js]: is this same test below?

File: index.test.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+		expect(filterOrange(arr, argument)).toContain('orange');
```
Comment (kenmori) [inline index.test.js]: this is your filterOrange function

```js
export function filterOrange(arr) {
	return arr.filter((e) => e === 'orange');
}
```
it dosen't recieve parameter as second argument, but you are passing it.

you should test for function that expect return `["orange"]`.

so just write, 

```suggestion
		expect(filterOrange(arr)).toStrictEqual(['orange']);
```

`toStrictEqual` often use to test object. [check more](https://jestjs.io/docs/expect#tostrictequalvalue)

File: index.test.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+	//toMatch toBe
+
+	test('add a + 1 to match a1', () => {
+		expect(addString('a', 1)).toBe('a1');
+	});
+
+	// if = false else return -1
+	test('add 1 + 1 return -1', () => {
+		expect(addString(1, 1)).toEqual(-1);
```
Comment (kenmori) [inline index.test.js]: please explain different between `toMatch` and `toBe` and `toEqual`

File: index.test.js (hunk 1 above)
Comment (nyattsu72) [reply index.test.js]: Thank you reviw！
Removed module loading.

[df70600](https://github.com/nyattsu72/handson-jest/pull/1/commits/df706009b40869f10bdf3263c4101daad810a347)

File: index.test.js (hunk 2 above)
Comment (nyattsu72) [reply index.test.js]: Thank you reviw！
Removed module loading.

[df70600](https://github.com/nyattsu72/handson-jest/pull/1/commits/df706009b40869f10bdf3263c4101daad810a347)

File: index.test.js (hunk 3 above)
Comment (nyattsu72) [reply index.test.js]: Thank you for revw!
Fixed orange text as orange element.

[df70600](https://github.com/nyattsu72/handson-jest/pull/1/commits/df706009b40869f10bdf3263c4101daad810a347)

File: index.test.js (hunk 4 above)
Comment (nyattsu72) [reply index.test.js]: Yes. I confirmed that the variable contains orange.
But,I misunderstood what to return with return.
This delete it.

[1a744fa](https://github.com/nyattsu72/handson-jest/pull/1/commits/1a744fa90752a3bd4e2adbc90a031928c596fe80)

File: index.test.js (hunk 5 above)
Comment (nyattsu72) [reply index.test.js]: Thank you for your review.
I misunderstood what to return with return.

I modified it to the matcher ".toStrictEquol" that you taught me.
[1a744fa](https://github.com/nyattsu72/handson-jest/pull/1/commits/1a744fa90752a3bd4e2adbc90a031928c596fe80)

File: index.test.js (hunk 6 above)
Comment (nyattsu72) [reply index.test.js]: Thank you for reviw！
There are three differences.

.toMatch = Check the character string with a regular expression. 

.toBe = Test for strict equivalence. Compare numbers and strings.
             ex)'1' === 1 /false    number === string

.toEqual = Compare the values ​​of objects. 
                 [aaaa, eeee]  {a:01 b:02}

If the result is a string, I modified it to .toMatch.
If the value changes, I modified it to .toBe.

[1a744fa](https://github.com/nyattsu72/handson-jest/pull/1/commits/1a744fa90752a3bd4e2adbc90a031928c596fe80)

File: index.test.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import { add, addString, toNumber, filterOrange } from './index';
+
+test('add 1 + 2 to equal 3', () => {
+	expect(add(undefined, 2)).toBe(NaN);
+});
+
```
Comment (kenmori) [inline index.test.js:6]: add normal test like below

```js
describe('if pass 1, then return 2', () => {
  test()
 });
```

File: index.test.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+describe('toNumber', () => {
+	//Check if it is a number
+	test('check a number', () => {
+		const string = 'string';
+		expect(toNumber(string).toNaN);
+	});
+
+	//Check if it text
```
Comment (kenmori) [inline index.test.js]: this comment no need
it you should write to describe function as first argument.

File: index.test.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+	});
+
+	// if = false else return -1
+	test('add 1 + 1 return -1', () => {
+		expect(addString(1, 1)).toBe(-1);
+	});
+});
+
+describe('toNumber', () => {
+	//Check if it is a number
```
Comment (kenmori) [inline index.test.js]: this comment no need
it you should write to describe function as first argument.

File: index.test.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+describe('addString', () => {
+	test('add a + b to match ab', () => {
+		expect(addString('a', 'b')).toMatch('ab');
+	});
+
+	test('add a + 1 to match a1', () => {
+		expect(addString('a', 1)).toMatch('a1');
+	});
+
+	// if = false else return -1
```
Comment (kenmori) [inline index.test.js]: this comment no need
it you should write to describe function as first argument.

File: index.test.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+test('add 1 + 2 to equal 3', () => {
+	expect(add(undefined, 2)).toBe(NaN);
+});
+
+//Task01
+/*
+・aとbが"a"、"b"の時"ab"となる
+・aが"a", bが1の時"a1"が返る
+・aが1、bが1の時-1が返る
+*/
```
Comment (kenmori) [inline index.test.js]: remove

File: index.test.js [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+
+describe('addString', () => {
+	test('add a + b to match ab', () => {
+		expect(addString('a', 'b')).toMatch('ab');
+	});
+
+	test('add a + 1 to match a1', () => {
+		expect(addString('a', 1)).toMatch('a1');
+	});
```
Comment (kenmori) [inline index.test.js]: write with discribe function

explain how to discribe and test function

Comment (nyattsu72) [reply index.test.js:20]: I'm sorry I forgot to delete the comment.
I deleted the comment.

[8ac52e7](https://github.com/nyattsu72/handson-jest/pull/1/commits/8ac52e7dc0a88abf57612ac6dbbc2de89316ef00)

File: index.test.js (hunk 11 above)
Comment (nyattsu72) [reply index.test.js]: I'm sorry I forgot to delete the comment.
I deleted the comment.

[8ac52e7](https://github.com/nyattsu72/handson-jest/pull/1/commits/8ac52e7dc0a88abf57612ac6dbbc2de89316ef00)

File: index.test.js (hunk 10 above)
Comment (nyattsu72) [reply index.test.js]: I'm sorry I forgot to delete the comment.
I deleted the comment.

[8ac52e7](https://github.com/nyattsu72/handson-jest/pull/1/commits/8ac52e7dc0a88abf57612ac6dbbc2de89316ef00)

File: index.test.js (hunk 9 above)
Comment (nyattsu72) [reply index.test.js]: I'm sorry I forgot to delete the comment.
I deleted the comment.

[8ac52e7](https://github.com/nyattsu72/handson-jest/pull/1/commits/8ac52e7dc0a88abf57612ac6dbbc2de89316ef00)

File: index.test.js (hunk 8 above)
Comment (nyattsu72) [reply index.test.js]: I'm sorry I forgot to delete the comment.
I deleted the comment.

[8ac52e7](https://github.com/nyattsu72/handson-jest/pull/1/commits/8ac52e7dc0a88abf57612ac6dbbc2de89316ef00)

File: index.test.js (hunk 7 above)
Comment (nyattsu72) [reply index.test.js:6]: Thank you for your new challenge.
I answered here.

[a6ab08b](https://github.com/nyattsu72/handson-jest/pull/1/commits/a6ab08b7a729a127da9a110528fbbc8940b29be5)

File: index.test.js (hunk 12 above)
Comment (nyattsu72) [reply index.test.js]: Thank you for your polite explanation on discord.

I found that describe( can be nested and used.
This was also officially written.
I rewrote the test again.

[67fb347](https://github.com/nyattsu72/handson-jest/pull/1/commits/67fb34741da58f18fe1d44bcfe8e2f5c6018bc41)

**Review by terracetech — APPROVED** [review]

**Comment by kenmori** [comment]
Approve(terracetechでapproveしてしまった)

**Pull request merged**